ext/opentelemetry-ext-wsgi/src/opentelemetry/ext/wsgi/flask_util.py

```diff
@@ -0,0 +1,88 @@
+# Note: This package is not named "flask" because of
+# https://github.com/PyCQA/pylint/issues/2648
+
+import logging
+
+from flask import request as flask_request
+
+import opentelemetry.ext.wsgi as otel_wsgi
+from opentelemetry import propagators, trace
+from opentelemetry.util import time_ns
+
+logger = logging.getLogger(__name__)
+
+_ENVIRON_STARTTIME_KEY = object()
+_ENVIRON_SPAN_KEY = object()
+_ENVIRON_ACTIVATION_KEY = object()
+
+
+def wrap_flask(flask):
+    wsgi = flask.wsgi_app
+
+    def wrapped_app(environ, start_response):
+        # We want to measure the time for route matching, etc.
+        # In theory, we could start the span here and use update_name later
+        # but that API is "highly discouraged" so we better avoid it.
+        environ[_ENVIRON_STARTTIME_KEY] = time_ns()
+
+        def _start_response(status, response_headers, *args, **kwargs):
+            span = flask_request.environ.get(_ENVIRON_SPAN_KEY)
+            if span:
+                otel_wsgi.add_response_attributes(
+                    span, status, response_headers
+                )
+            else:
+                logger.warning(
+                    "Flask environ's OTel span missing at _start_response(%s)",
```

Comment:
In error messages, I feel like we should avoid abbreviation. not that OTel is that hard to correlate, but I'd much rather have it immediately be clear to the user it's OpenTelemetry related rather than have to google what "OTel" is.

Reply:
👍 , although users would be well-advised to configure logging so that they now that from the logger name.

Reply:
Done.